shotover-proxy/src/transforms/protect/crypto.rs

```diff
@@ -53,10 +51,10 @@ pub async fn decrypt(
     key_id: &str,
 ) -> Result<MessageValue> {
     let varchar = match value {
-        MessageValue::Varchar(varchar) => varchar,
+        MessageValue::Bytes(varchar) => varchar,
         _ => bail!("expected varchar to decrypt but was {:?}", value),
     };
-    let protected: Protected = serde_json::from_str(varchar)?;
+    let protected: Protected = bincode::deserialize(&hex::decode(varchar)?)?;
```

Comment:
the hex::decode needs to be removed here.
Cassandra results include the bytes directly as Bytes.